cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java

```diff
@@ -326,46 +327,52 @@ public Future<CruiseControlResponse> getUserTaskStatus(String host, int port, St
                                 String userTaskID = response.result().getHeader(USER_TASK_ID_HEADER);
                                 response.result().bodyHandler(buffer -> {
                                     JsonObject json = buffer.toJsonObject();
-                                    JsonObject jsonUserTask = json.getJsonArray("userTasks").getJsonObject(0);
-                                    // This should not be an error with a 200 status but we play it safe
-                                    if (jsonUserTask.containsKey(CC_REST_API_ERROR_KEY)) {
-                                        result.fail(new CruiseControlRestException(
-                                                "Error for request: " + host + ":" + port + path + ". Server returned: " +
-                                                        json.getString(CC_REST_API_ERROR_KEY)));
-                                    }
```

Comment:
What if this is some different error? Do we really want to loop on every single error?

Reply:
It was something present there since the start, This condition is just to be sure nothing went wrong even after we got a 200 status response. I am not sure what that edge case could be but I guess its good to have something to tackle it

Reply:
Also as you said `what if there is some different error`,  I think thats a good point but if there is CC related error and status code is 200 then it should be present in the `json` with the `CC_REST_API_ERROR_KEY`.